src/logicTemplates/htlc.js

```diff
@@ -67,6 +67,26 @@ class HTLC {
     }
 }
 
+/* signTransactionWithHTLCUnlock accepts a transaction, such as a payment, and builds the HTLC-unlocking signature around that transaction
+* @param {Uint8Array} byte representation of the HTLC
+* @param {Object} txn dictionary containing constructor arguments for a transaction
+* @param {string} preimage of the hash as base64 string
+*
+* @returns {Object} Object containing txID and blob representing signed transaction.
+ */
+function signTransactionWithHTLCUnlock(contract, txn, preImageAsBase64) {
```

Comment:
Missing some validation:
* hash function (`program[len(program)-15]`), you can check the java/python implementations for this.
* max fee (`ints[1]`)

The python/java implementations also construct the transaction in here instead of accepting a transaction. Which do you think is better?

Reply:
There's been UX problems in the javascript SDK before with internally constructing transactions, I'd prefer accepting a whole dict over the individual arguments. Max fee and hash function validation added.